crates/ty_server/src/server/main_loop.rs

```diff
@@ -113,8 +141,8 @@ impl Server {
     /// Returns `Ok(None)` if the client connection is closed.
     fn next_event(&self) -> Result<Option<Event>, crossbeam::channel::RecvError> {
         let next = select!(
```

Comment:
The variable assignment seems unnecessary as the `next.map(Some)` is only for a single branch, maybe we can simplify it:

```rs
        select!(
            recv(self.connection.receiver) -> msg => Ok(msg.ok().map(Event::Message)),
            recv(self.main_loop_receiver) -> event => event.map(Some),
        )
```